tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -4246,6 +4247,70 @@ void CGMSHLSLRuntime::SetPatchConstantFunctionWithAttr(
   
 }
 
+// Returns true a global value is being updated
+static bool IsWriteEnabledGlobalRec(Value *V, std::set<Value*> &visited) {
+  bool isWriteEnabled = false;
+  if (V && visited.find(V) == visited.end()) {
+    visited.insert(V);
+    for (User *U : V->users()) {
+      if (isa<StoreInst>(U))
+        isWriteEnabled = true;
+      else if (GetElementPtrInst *GEP = dyn_cast<GetElementPtrInst>(U)) {
```

Comment:
This can be a constant GEP expression, so best to use GEPOperator.  Further, you can have multiple levels of GEP at this point, so best to recurse.  In fact, this means you don't need the local variable, since you just return true when you find a store, or when a recursive call returns true.  Then return false at the end, when you didn't find a store.